### PR TITLE
User registration

### PR DESCRIPTION
**Pull request by hashmaster3k**
### Description:

  Visitors can now register and log in.

### Tasks:

- [x] Email
- [x] Password
- [x] Password Confirmation
- [x] Register Button

#### Notes:
  Tested edge casing for uniqueness of email, and correct credentials.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,10 +16,12 @@ gem 'coffee-rails', '~> 4.2'
 gem 'bootsnap'
 gem 'jbuilder', '~> 2.5'
 gem 'rubocop', require: false
+gem 'bcrypt'
```
Comment (hashmaster3k) [inline Gemfile:19]: Used for password digest

File: Gemfile [hunk 2]
```diff
@@ -16,10 +16,12 @@ gem 'coffee-rails', '~> 4.2'
...
 
 group :development, :test do
   gem 'pry'
   gem 'travis'
+  gem 'shoulda-matchers'
```
Comment (hashmaster3k) [inline Gemfile:24]: Use for testing validations on user model.

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        redirect_to '/user/dashboard'
+      else
+        flash[:error] = "Credentials are incorrect"
+        redirect_to '/'
+      end
+    else
+      flash[:error] = "Credentials are incorrect"
+      redirect_to '/'
+    end
+  end
```
Comment (hashmaster3k) [inline app/controllers/sessions_controller.rb:15]: A little lenghty. Forgot to run RuboCop to refactor into smaller method.

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  def create_user
+    user = User.new(user_params)
+    if user.save
+      session[:user_id] = user.id
+      redirect_to '/user/dashboard'
+    else
+      flash[:error] = user.errors.full_messages.first
+      render :new
+    end
+  end
```
Comment (hashmaster3k) [inline app/controllers/users_controller.rb:24]: Refactored the actual user creation into helper method.

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<center>
+<h1>Welcome to Viewing Party</h1>
+
+<p>Please Register for an Account</p>
+
+<div id="registration">
+  <%= form_with url: "/registration", local: true do |form| %>
```
Comment (hashmaster3k) [inline app/views/users/new.html.erb:7]: Would be a good idea to refactor all paths to using routes pathing nomenclature.

File: app/controllers/sessions_controller.rb (hunk 3 above)
Comment (nicorithner) [reply app/controllers/sessions_controller.rb:15]: We can refactor that on next branch

**Pull request merged**